lib/hooks.ts

```diff
@@ -30,24 +31,28 @@ export function useProducts() {
     };
 }
 
-export function useProductList() {
-    const encodedContext = useSession()?.context;
+export function useProductList(query?: QueryProps) {
+    const { context } = useSession();
+    const params = new URLSearchParams({ ...query, context }).toString();
+
     // Use an array to send multiple arguments to fetcher
-    const { data, error, mutate: mutateList } = useSWR(encodedContext ? ['/api/products/list', encodedContext] : null, fetcher);
+    const { data, error, mutate: mutateList } = useSWR(context ? ['/api/products/list', params] : null, fetcher);
 
     return {
-        list: data,
+        list: data?.data,
+        meta: data?.meta,
         isLoading: !data && !error,
         error,
         mutateList,
     };
 }
 
 export function useProductInfo(pid: number, list: ListItem[]) {
-    const encodedContext = useSession()?.context;
+    const { context } = useSession();
+    const params = new URLSearchParams({ context: context }).toString();
```

Comment:
if the object property name is equal to a variable, we should use the [shorthand property name](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer#new_notations_in_ecmascript_2015). DRY
```suggestion
    const params = new URLSearchParams({ context }).toString();
```